pipeline/common/downloads.py

```diff
@@ -471,3 +471,81 @@ def write_lines(path: Path | str):
 
     finally:
         stack.close()
+
+
+def compress_file(
```

Comment:
Should we add unit tests for compress/decompress functions? They are quite important to work correctly now.

Reply:
Good idea. I'll do it. I think we're covered by integration tests, but it's always nice to be explicit on stuff like this.